### PR TITLE
Moronic ui

### PR DESCRIPTION
**Pull request by bion**

### CODE REVIEW

File: src/js/components/midi_range_selector.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class MIDIRangeSelector extends Component {
+  render() {
+    return (
+      <div className="range-selector">
+        <label htmlFor={this.props.path}>{this.props.name}</label>
+        <input
+          type='number'
+          ref='input'
+          name={this.props.path}
+          min='0'
```
Comment (dbalatero) [inline src/js/components/midi_range_selector.js]: I'd make the (min, max) a prop that defaults to 0 and 127?

File: src/js/components/midi_range_selector.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React, { PropTypes } from 'react';
```
Comment (bion) [inline src/js/components/midi_range_selector.js]: @dbalatero 

:information_desk_person: babel generates something that looks like this whenever it sees jsx (i.e. XML acting like it belongs in JS) code:

```
(0, _reactDom.render)(React.createElement(
  _reactRedux.Provider,
  { store: store },
  React.createElement(_patch_app2.default, null)
```

pretty neat. this means that we have to `import React from 'react'` like this in any file dealing with with code that will be treated as a React component (even if it doesn't inherit explicitly from Component and is just a markup returning function, like on line 4 down there).

i think this is silly and we should consider using https://github.com/re5et/jsxon

File: src/js/reducers.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  }
+};
+
+function PatchAppReducer(state = initialState, action) {
+  switch (action.type) {
+  case 'SET_PARAM':
+    return Object.assign({}, state, {
+      currentPatch: {
+        state: {
+          [action.path]: action.value
```
Comment (bion) [inline src/js/reducers.js]: @dbalatero scope this [computed property name](http://es6-features.org/#ComputedPropertyNames) action

**Comment by dbalatero** [comment]
@bion ah you finished it then?

**Comment by bion** [comment]
@dbalatero just the one control value

**Pull request merged**